components/automate-deployment/pkg/backup/artifact_repo.go

```diff
@@ -432,9 +650,65 @@ func (repo *ArtifactRepo) openSnapshotFile(ctx context.Context, name string) (Ar
 	return NewLineReaderStream(tmpFile), checksum, nil
 }
 
+func (repo *ArtifactRepo) openSnapshotFile(ctx context.Context, name string) (ArtifactStream, string, error) {
+	return repo.openGzipFile(ctx, toSnapshotName(name))
+}
+
+// ReadSnapshotIntegrityMetadata reads the snapshot integrity metadata file
+func (repo *ArtifactRepo) ReadSnapshotIntegrityMetadata(ctx context.Context) (*ArtifactRepoIntegrityMetadata, error) {
+	metadata := NewArtifactRepoIntegrityMetadata()
+
+	metadataStream, _, err := repo.openGzipFile(ctx, SnapshotIntegrityMetadataFileName)
```

Comment:
I dont think artifact stream is the right thing to use here. The artifact streams are line oriented files that are used to deal with a large list of things. This should be reading the json.gz file. I'm sort of not clear why this even works as it would strip newlines and corrupt the gzip?

Reply:
It seems to be because LoggingStream writes line oriented files